pom.xml

```diff
@@ -503,37 +508,6 @@
                     <version>1.3</version>
                 </plugin>
                 <plugin>
-                    <groupId>org.bsc.maven</groupId>
```

Comment:
> What's the reason for removing this? 
Several reasons:
* the long outstanding upgrade of JBoss Logging
* letting the annotation processor play "nicer" with the IDE. In its current setup it is not working from Idea and one needs to manually re-create the Log class on changes to the interface
* according to David this is the "recommended" setup

>  I liked the fact that this set-up does not add the processor as normal project dependency but only on the separate processor path

Me too. As you know I was arguing for it for a long time, even refusing an earlier upgrade of JBoss Logging. Since other principles already weakened (see Eclipse pulgin settings), I think this is another logical step. It makes working with the IDE easier and we made that now an important goal.

> There is a separate artifact with only the annotations which can be used as project dependency.

That's not helping the goal of letting the processor run as part of the compile step.

Btw, as a final step I am considering a dedicates testutil module as mentioned in other comments.


I liked the fact that this set-up does not add the processor as normal project dependency but only on the separate processor path. There is a separate artifact with only the annotations which can be used as project dependency.